sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/WriteOutput.scala

```diff
@@ -375,6 +378,7 @@ object WriteOutput extends Logging {
 
       // If anything below fails, we should abort the task.
       var currentKey: UnsafeRow = null
+      var updatedPartitions: List[String] = Nil
```

Comment:
In case of bucketing, There are multiple files (writer) per partition, so partitionPath will have duplicated value, should we use Set[String] here?